assets/js/phoenix_live_view/rendered.js

```diff
@@ -121,23 +217,40 @@ export default class Rendered {
         target[key] = val
       }
     }
+    if(target[ROOT]){
+      target.newRender = true
+    }
   }
 
-  cloneMerge(target, source){
+  // Merges cid trees together, copying statics from source tree.
+  //
+  // The `pruneMagicId` is passed to control pruning the magicId of the
+  // target. We can only prune the magicId if we are merging into a target
+  // the first time, otherwise we need to maintain the magicId and set newRender
+  // to true so the next render won't be skipped.
```

Comment:
```suggestion
  // The `pruneMagicId` is passed to control pruning the magicId of the
  // target. We must always prune the magicId when we are sharing statics
  // from another component. If not pruning, we replicate the logic from
  // mutableMerge, where we set newRender to true if there is a root
  // (effectively forcing the new version to be rendered instead of skipped)
```